diff --git a/src/debug_menu.cpp b/src/debug_menu.cpp
index d9aa59fe82..d0e9fd8347 100644
--- a/src/debug_menu.cpp
+++ b/src/debug_menu.cpp
@@ -437,12 +437,7 @@ static cata::optional<debug_menu_index> debug_menu_uilist( bool display_all_entr
         menu.insert( menu.begin() + 1, debug_menu.begin(), debug_menu.end() );
     }
 
-    std::string msg;
-    if( display_all_entries && !is_debug_character() ) {
-        msg = _( "Debug Functions - Using these will cheat not only the game, but yourself.\nYou won't grow.  You won't improve.\nTaking this shortcut will gain you nothing.  Your victory will be hollow.\nNothing will be risked and nothing will be gained." );
-    } else {
-        msg = _( "Debug Functions" );
-    }
+    std::string msg = _( "Debug Functions" );
 
     while( true ) {
         const int group = uilist( msg, menu );
