From 3deff03ff9187fff04d6fe604ad007fae7ba0019 Mon Sep 17 00:00:00 2001
From: RenechCDDA <84619419+RenechCDDA@users.noreply.github.com>
Date: Mon, 5 Aug 2024 10:27:28 -0400
Subject: [PATCH 073/125] get_driver can return nullptr

---
 src/vehicle.cpp | 12 +++++++++---
 1 file changed, 9 insertions(+), 3 deletions(-)

diff --git a/src/vehicle.cpp b/src/vehicle.cpp
index 13665d5654..e22ecc7cb2 100644
--- a/src/vehicle.cpp
+++ b/src/vehicle.cpp
@@ -3442,7 +3442,14 @@ bool vehicle::has_driver() const
 
 Character *vehicle::get_driver() const
 {
-    return &get_player_character();
+    // TODO: Gotta be a better way than this...
+    for( const vpart_reference &vp : get_all_parts() ) {
+        Character *occupant = vp.get_passenger();
+        if( occupant && player_in_control( *occupant ) ) {
+            return occupant;
+        }
+    }
+    return nullptr;
 }
 
 monster *vehicle::get_monster( int p ) const
@@ -3602,8 +3609,7 @@ int64_t vehicle::fuel_left( const itype_id &ftype,
 
         //if the engine in the tile is a muscle engine, and someone is ready to control this vehicle
         // TODO: Allow NPCs to power those
-        // TODO: Revise this check to driver when get_driver() can return nullptr
-        if( vp && &vp->vehicle() == this && player_is_driving_this_veh() ) {
+        if( vp && &vp->vehicle() == this && driver ) {
             const int p = avail_part_with_feature( vp->part_index(), VPFLAG_ENGINE );
             if( p >= 0 ) {
                 const vehicle_part &vp = parts[p];
-- 
2.45.2

