From 6a8322b0d242d2e74397c2b1c914d2d643c86ef7 Mon Sep 17 00:00:00 2001
From: RenechCDDA <84619419+RenechCDDA@users.noreply.github.com>
Date: Tue, 13 Aug 2024 11:57:25 -0400
Subject: [PATCH 076/125] Silly override hack for tests

---
 src/vehicle.cpp | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/src/vehicle.cpp b/src/vehicle.cpp
index bcb7103238..a0cbae25fa 100644
--- a/src/vehicle.cpp
+++ b/src/vehicle.cpp
@@ -273,6 +273,12 @@ bool vehicle::player_in_control( const Character &p ) const
 
 bool vehicle::player_is_driving_this_veh() const
 {
+    // Unfortunate code duplication for tests
+    // Debug switch to prevent vehicles from skidding
+    // without having to place the player in them.
+    if( tags.count( "IN_CONTROL_OVERRIDE" ) ) {
+        return true;
+    }
     // Early out, nobody's driving
     if( !get_driver() ) {
         return false;
-- 
2.45.2

